### PR TITLE
More parsing/printing notation/abbreviation consistency for mixed terms and pattern

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** bug fix / cleanup

Depends on #8808 (now merged).

This is a continuation of #8808 about mixed binders and terms in notations and abbreviations. It ensures consistency of printing and parsing for both notations and abbreviations in the different possible configurations (see test-suite file `Notations4.v`).

For the record, the typical application of such mixed terms and binders notations is the ["maximal group such that"](https://github.com/math-comp/math-comp/blob/732dc474f09c0231e2332cdecf99a3ed045cdd04/mathcomp/fingroup/fingroup.v#L3060) used in MathComp or the "forcing" modality:
```
Definition force n (P:nat -> Prop) := forall n', n' >= n -> P n'.
Notation "▢_ n P" := (force n (fun n => P))
   (at level 0, n ident, P at level 9, format "▢_ n  P").
Check exists p, ▢_p (p >= 1).
```

It does not fix the printing order issue that #9311 is trying to fix.

- [X] Added / updated test-suite
Corresponding documentation was added in #8808.
- [x] Entry added in the changelog (this is not new, but the advertising is new; note that a word was already said in #8808; so maybe this is overkill).

### CODE REVIEW

**Comment by Zimmi48** [comment]
I'm turning this PR in draft mode because of the dependency. If you expect feedback from some specific groups of maintainers, please say so. With PR dependencies, try to always use the draft mechanism, otherwise the code owner mechanism is generating review spam.

**Comment by Zimmi48** [comment]
Dear PR author(s) and contributors, it seems to us this PR may need additional work beyond the 8.12 branching deadline on May 15th.

We are thus removing the 8.12+beta1 milestone; please retarget as appropriate, and also consider updating labels to reflect current status, etc.

We apologize in advance if we misunderstood the PR status.

**Comment by herbelin** [comment]
It depends on #8808 which depends on @mattam82 who seems to have other priorities these days. I just asked @ejgallego instead.

**Comment by herbelin** [comment]
I just found the time to rebase this PR after #8808 is merged. It is basically a refinement of #8808, see the `▢_ n P` notation example.

**Comment by herbelin** [comment]
@ejgallego: Is it asking a lot to possibly review this PR?

**Comment by herbelin** [comment]
Rebased and dependencies merged.

**Comment by ejgallego** [comment]
@herbelin I'm reviewing, sadly the CI didn't run due to the limits, please rebase and push.

**Comment by ejgallego** [comment]
@herbelin unfortunately this got a conflict :S please rebase, and hopefully we can merge.

**Comment by herbelin** [comment]
OK, rebased.

**Comment by ejgallego** [comment]
Failure are unrelated, merging.

@coqbot: merge now

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@coqbot: merge now

**Comment by ejgallego** [comment]
@coqbot: merge now (sudo)

**Pull request merged**
